nodebuilder/das/das.go

```diff
@@ -17,6 +19,16 @@ type Module interface {
 // API is a wrapper around Module for the RPC.
 // TODO(@distractedm1nd): These structs need to be autogenerated.
 type API struct {
-	SamplingStats func(ctx context.Context) (das.SamplingStats, error)
-	WaitCatchUp   func(ctx context.Context) error
+	Internal struct {
+		SamplingStats func(ctx context.Context) (das.SamplingStats, error) `perm:"read"`
+		WaitCatchUp   func(ctx context.Context) error                      `perm:"write"`
```

Comment:
Tend to agree that this should be `write`, but still curios on your rationale

Reply:
Hmm. Yeah `WaitCatchUp` definitely seems like a write.
On the fence about `SamplingStats`, I can see the case for both read and write. Would love to hear from @walldiss

Reply:
`WaitCatchUp` will block client until catchup happens. Both methods don't mutate any data, so there is nothing to write, unless I misunderstand concept of `write` permission.

Reply:
I saw this as a long-blocking operation that might have. Nothing to write but also nothing to read. Such blocking operations have DOS implications, so that I would hide them away(but that might be overcautious, as every single read op have similar issues) Although, read is fine if apply suggestion with separate `public` perm which SGTM

Reply:
Oh, it is suppose to be a subscription, but with rpc it turned into long-living rpc call.  I'm not even sure we should expose it this way. At very least we should be careful a timeout on such call.

Reply:
We can expose it so that our CLI can block and wait until catching up is done, but that does not require us to have full public perms and can be something private.

Reply:
both methods are behind read now